Add .zfill() Method Entry in 'Terms' Section (#3367) (#3447)
Codecademy/docs: content/python/concepts/strings/terms/zfill/zfill.md

---
Title: '.zfill()' 
Description: 'Takes an int and returns a copy of the string with a number of zeros preppended as a prefix , i.e "no. of zeros" + string.'
Subjects:
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Methods'
  - 'Functions'
CatalogContent:
  - 'learn-python-3'
  - 'paths/analyze-data-with-python'
---

The **`.zfill()`**  method pads a string with zeros on the left to maintain a specific length. It takes an integer argument, and the number of zeros added is determined by the difference between the specified length and the length of the original string.

> **Note:** The `.zfill()` method does not change the string it is used on.

## Syntax

```pseudo
result = string.zfill(arg(int))
```

> It creates a new string with leading zeros based on the specified integer argument, represented by `arg`. The modified string is stored in the variable `result`.

Reviewer: ```suggestion
It creates a new string with leading zeros based on the specified integer argument, represented by `total_length`. The modified string is stored in the variable `result`.
```

## Example

The following example shows how `.zfill()` returns a copy of the string with leading zeros. 

```py
first_string = "Codecademy"
second_string = "@Codecademy"
third_string = "cat"

print(first_string)
print(second_string)

> # parameter is less than/ equal to string length, output will be the same string.

Reviewer: ```suggestion
# If the argument is less than/equal to string length, output will be the same string.
```

print(first_string.zfill(10))
print(second_string.zfill(10))

print(first_string.zfill(12))
print(second_string.zfill(12))
print(third_string.zfill(12))

> ```

Reviewer: There should be an empty line after the code block

This would output:

```shell
Codecademy
@Codecademy

Codecademy
@Codecademy

00Codecademy
0@Codecademy
000000000cat

> 

Reviewer: ```suggestion
```

---
Title: '.zfill()' 

> Description: 'Takes an int and returns a copy of the string with a number of zeros preppended as a suffix , i.e "no. of zeros" + string.'

Reviewer 2: ```suggestion
Description: 'Takes an int and returns a copy of the string with a number of zeros prepended as a prefix, i.e "no. of zeros" + string.'
```

Subjects:
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Methods'
  - 'Functions'
CatalogContent:
  - 'learn-python-3'
  - 'paths/analyze-data-with-python'
---

The **`.zfill()`**  method pads a string with zeros on the left to maintain a specific length. It takes an integer argument, and the number of zeros added is determined by the difference between the specified length and the length of the original string.

> > **Note:** The `.zfill()` method does not change the string it is used on.
> If no argument is passed into the method. A type error is thrown

Reviewer 2: ```suggestion
> **Note:** The `.zfill()` method does not change the string it is used on.
```
We don't need the line about the error, just note above that the parameter is required.

> ## Syntax
> ```pseudo
> string.zfill(arg(int))
> ```

Reviewer 2: ````suggestion
## Syntax

```pseudo
result = string.zfill(arg)
```
````
Also add a sentence after the code block explaining what `result` and `arg` are.

> ## Codebyte Example
> The following example shows how `.zfill()` returns a copy of that string with leading zeros. 

Reviewer 2: ```suggestion
## Example 1

The following example shows how `.zfill()` returns a copy of that string with leading zeros: 
```

```py
first_string = "Codecademy"
second_string = "@Codecademy"
third_string = "cat"

print(first_string)
print(second_string)

# parameter is less than/ equal to string length, output will be the same string.
print(first_string.zfill(10))
print(second_string.zfill(10))

print(first_string.zfill(12))
print(second_string.zfill(12))
print(third_string.zfill(12))
```
This would output:

```shell
Codecademy
@Codecademy

Codecademy
@Codecademy

00Codecademy
0@Codecademy
000000000cat

```

> ## Example

Reviewer 2: This section is really not necessary. All functions with required parameters will throw an error if the parameter is missing. It isn't something specific to `.zfill()`

The following example shows the error thrown when `.zfill()` is used without an arguement passed in

```py
string = "Codecademy"

string.zfill()
```

This would output:

```shell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: zfill() takes exactly 1 argument (0 given)
```

The above error simply means the `.zfill` method is missing the integer paramater

> ## Codebyte Example
> The following example shows how `.zfill()` method helps ensure a specific length for a string, adding zeros to the beginning if the string is shorter than the desired length.

Reviewer 2: ```suggestion
## Codebyte Example

The following example shows how `.zfill()` method helps ensure a specific length for a string, adding zeros to the beginning if the string is shorter than the desired length.
```
